tablib/core.py

```diff
@@ -452,6 +482,164 @@ def load(self, in_stream, format=None, **kwargs):
+        :param font_color: fonr color, str
+        :param border: border index, int
+        :param bold: bool
+        :param italic: bool
+        :param aligment: text position in the cell, str
+        :return: None
+        """
+        column_position = self._detect_column(column)
+        self._set_format(column_position, None, bg_color, font, font_size, font_color, border, bold, italic, aligment)
+
+    def format_header(self, bg_color=None, font=None, font_size=None, font_color=None, border=None, bold=False,
+                   italic=False, aligment=None):
+        """
+         Applying format to the header
+         :param bg_color: cell backgroud color, str
+         :param font: font name, str
+         :param font_size: font size, int
+         :param font_color: fonr color, str
+         :param border: border index, int
+         :param bold: bool
+         :param italic: bool
+         :param aligment: text position in the cell, str
+         :return: None
+         """
+        self._set_format(None, 0, bg_color, font, font_size, font_color, border, bold, italic, aligment)
+
+    def format_footer(self, columns, bg_color=None, font=None, font_size=None, font_color=None, border=None, bold=False,
+                   italic=False, aligment=None):
+        """
+         Applying format to the footer
+         :param bg_color: cell backgroud color, str
+         :param font: font name, str
+         :param font_size: font size, int
+         :param font_color: fonr color, str
+         :param border: border index, int
+         :param bold: bool
+         :param italic: bool
+         :param aligment: text position in the cell, str
+         :return: None
+         """
+        self.__footer = columns
+        self.__footer_format = self._make_format(bg_color, font, font_size, font_color, border, bold, italic, aligment)
+
+
+    def add_dropdown(self, column, source):
+        """
+        Add dropdown to the cell
+        :param column: column name, str
+        :param source: choices for dropdown, list
+        :return: None
+        """
+        column_position = self._detect_column(column)
+        self._set_dropdown(1, column_position, len(self._package(dicts=False)),
+                           column_position, {'validate': 'list', 'source': map(str, source)})
+
+    def add_conditional_formatting(self, conditions, column=None):
+        """
+        Add conditional format to the cell
+        :param conditions: conditions, list of the dicts
+        :param column: column name, str
+        :return: None
+        """
+        column_position = self._detect_column(column)
+        for condition in conditions:
+            cell_format = {'bg_color': condition.get('color', None)}
+            value = '"{}"'.format(condition['value']) if isinstance(condition['value'], str) else condition['value']
+            self._set_conditional_format(1, column_position, len(self._package(dicts=False)) - 1,
```

Comment:
We should avoid generating `_package` here - it's an expensive operation.
`self.height` should be enough.

That said, it might be better to avoid height entirely.
What happens if you add more columns after this conditional formatting?
I think they don't get the conditional formatting applied.
It would be more robust to record the column only.

Reply:
done